azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CFilterScenario.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.autoconfigure.b2c;
+
+import lombok.Getter;
+import lombok.NonNull;
+import org.springframework.http.HttpMethod;
+
+import javax.servlet.http.HttpServletRequest;
+
+public enum AADB2CFilterScenario {
+
+    SIGN_IN_OR_SIGN_UP(new AADB2CFilterSignUpOrInHandler()),
+    LOGOUT_SUCCESS(new AADB2CFilterLogoutSuccessHandler()),
+    DEFAULT_SCENARIO(new AADB2CFilterDefaultHandler());
+
+    @Getter
+    private AADB2CFilterScenarioHandler scenarioHandler;
+
+    AADB2CFilterScenario(@NonNull AADB2CFilterScenarioHandler scenarioHandler) {
+        this.scenarioHandler = scenarioHandler;
+    }
+
+    public static AADB2CFilterScenario resolve(@NonNull HttpServletRequest request, @NonNull AADB2CFilter filter) {
+        final String requestURL = request.getRequestURL().toString();
+        final String method = request.getMethod();
+
+        if (HttpMethod.GET.matches(method) && requestURL.equals(filter.getSignUpOrInRedirectURL())) {
```

Comment:
Ignore case equals?

Reply:
No need here, as Method names are case-sensitive, `get` is another method like user defined.

Reply:
What about `requestURL`?

Reply:
In fact, only domain name is case insenstive, so we cannot do that here.
For example, 
https://en.wikipedia.org/wiki/Case_sensitivity
https://en.wikipedia.org/wiki/CASE_SENSITIVITY
Are totally different pages.